### PR TITLE
unify directives presentation

### PR DESCRIPTION
**Pull request by rsill-neo4j**
added a number of "tba" in comments. mostly missing definitions, but on some occasions i wasn't sure which parts of the follow-up sections could got to the "usage" subsection.

### CODE REVIEW

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-graphql-211.surge.sh

Updated pages:
- [Custom logic](https://neo4j-docs-graphql-211.surge.sh/graphql/5/directives/custom-logic)
- [Database mapping](https://neo4j-docs-graphql-211.surge.sh/graphql/5/directives/database-mapping)
- [Directives](https://neo4j-docs-graphql-211.surge.sh/graphql/5/directives)
- [Indexes and constraints](https://neo4j-docs-graphql-211.surge.sh/graphql/5/directives/indexes-and-constraints)
- [Field configuration](https://neo4j-docs-graphql-211.surge.sh/graphql/5/directives/schema-configuration/field-configuration)
- [Type configuration](https://neo4j-docs-graphql-211.surge.sh/graphql/5/directives/schema-configuration/type-configuration)
- [Relay Compatibility](https://neo4j-docs-graphql-211.surge.sh/graphql/5/integrations/relay-compatibility)
- [Authentication](https://neo4j-docs-graphql-211.surge.sh/graphql/5/security/authentication)
- [Authorization](https://neo4j-docs-graphql-211.surge.sh/graphql/5/security/authorization)

File: modules/ROOT/pages/directives/custom-logic.adoc
Comment (lidiazuin) [inline modules/ROOT/pages/directives/custom-logic.adoc:1]: Should `=== Context values` become `==== Context values` like in other cases such as "Return values" etc?

File: modules/ROOT/pages/directives/database-mapping.adoc [hunk 1]
```diff
@@ -10,10 +10,16 @@ Each type in your GraphQL type definitions can be mapped to an entity in your Ne
...
+
+// tba
+
+=== Usage
+
 Relationships are represented by marking particular fields with a directive -- in this case, `@relationship`. 
 It defines the relationship type in the database, as well as which direction that relationship goes in.
 
-To add a second node type, "Actor", and connect the two together, you should do the following:
+To add two node types, "Movie" and "Actor", and connect the two:
```
Comment (lidiazuin) [inline modules/ROOT/pages/directives/database-mapping.adoc:43]: clever solution!!

File: modules/ROOT/pages/directives/database-mapping.adoc [hunk 2]
```diff
@@ -28,8 +34,14 @@ type Actor {
 }
 ----
 
-Note that, in this case, there is a directive on each "end" of the relationship, but it is not essential.
+[NOTE]
+====
+The `@relationship` directive is used twice, once on each end of the relationship.
+This is the standard way of modeling a relationship with the GraphQL library.
+However, it is not a requirement of the type definitions themselves and relationships can deliberately be underspecified, for example to limit access through the API layer.
```
Comment (lidiazuin) [inline modules/ROOT/pages/directives/database-mapping.adoc]:
```suggestion
However, it is not a requirement of the type definitions themselves as relationships can deliberately be underspecified, for example to limit access through the API layer.
```

File: modules/ROOT/pages/directives/database-mapping.adoc [hunk 3]
```diff
@@ -28,8 +34,14 @@ type Actor {
...
+====
 
+ Also see xref::/directives/schema-configuration/field-configuration/#_relationship[`@relationship` field configuration].
```
Comment (lidiazuin) [inline modules/ROOT/pages/directives/database-mapping.adoc]: Weird... this is showing in a gray/code block.. might be a problem with the preview.

```suggestion
See also: xref::/directives/schema-configuration/field-configuration/#_relationship[`@relationship` field configuration].
```

File: modules/ROOT/pages/directives/indexes-and-constraints.adoc [hunk 4]
```diff
@@ -5,64 +5,10 @@
 
 This page describes how to use indexes and constraints in the Neo4j GraphQL Library.
 
-== Unique node property constraints
+[#_fulltext_indexes]
```
Comment (lidiazuin) [inline modules/ROOT/pages/directives/indexes-and-constraints.adoc]: Is there a reason why you are using this here? I think these paragraph-specific URLs are automatically generated after the TOC headers. 
If that's to make it a more specific URL, then we need to make sure they are updated elsewhere (I don't know if redirects work for paragraph-specific URLs, but I think they do).

File: modules/ROOT/pages/directives/indexes-and-constraints.adoc [hunk 5]
```diff
@@ -221,6 +165,66 @@ query {
 }
 ----
 
+[#_unique_node_property_constraints]
+== `@unique`
+
+=== Definition
+
+Unique node property constraints map to `@unique` directives used in your type definitions, which has the following definition:
```
Comment (lidiazuin) [inline modules/ROOT/pages/directives/indexes-and-constraints.adoc]:
```suggestion
Unique node property constraints map to `@unique` directives used in your type definitions.
They have the following definition:
```

File: modules/ROOT/pages/directives/indexes-and-constraints.adoc [hunk 6]
```diff
@@ -221,6 +165,66 @@ query {
...
+[source, graphql, indent=0]
+----
+"""Informs @neo4j/graphql that there should be a uniqueness constraint in the database for the decorated field."""
+directive @unique(
+    """The name which should be used for this constraint. By default; type name, followed by an underscore, followed by the field name."""
+    constraintName: String
+) on FIELD_DEFINITION
+----
+
+Using this directive does not automatically ensure the existence of these constraints, and you will need to run a function on server startup.
```
Comment (lidiazuin) [inline modules/ROOT/pages/directives/indexes-and-constraints.adoc]:
```suggestion
Using this directive does not automatically ensure the existence of these constraints.
Run a function on server startup.
```

edit by rsill-neo4j: shortened to imperative

File: modules/ROOT/pages/security/authorization.adoc [hunk 7]
```diff
@@ -7,9 +7,19 @@ They use predicates to evaluate the data accessed by the Cypher generated from a
 
 All authorization rules have an implied requirement for authentication, given that the rules are normally evaluated against values in the JWT payload.
 
-In the case of explicit authentication, configured using the `@authentication` directive, it is only ever evaluated during Cypher translation time.
+In the case of explicit authentication, configured using the xref::/security/authentication.adoc#_authentication[`@authentication`] directive, it is only ever evaluated during Cypher translation time.
```
Comment (lidiazuin) [inline modules/ROOT/pages/security/authorization.adoc]: Following the previous edit to a similar sentence:

```suggestion
In the case of explicit authentication, configured using the xref::/security/authentication.adoc#_authentication[`@authentication`] directive, it is only evaluated during Cypher translation time.
```

File: modules/ROOT/pages/security/subscriptions-authorization.adoc [hunk 8]
```diff
@@ -3,13 +3,22 @@
 = Subscriptions authorization
 
 Subscriptions require their own authorization rules, which are configured with the `@subscriptionsAuthorization` directive.
-These rules are different to normal authorization rules because only filtering rules are available for subscriptions events and subscriptions events have more limitations in how they can be filtered.
+
+== `@subscriptionsAuthorization`
+
+Rules for subscriptions authorization are different to normal authorization rules because only filtering rules are available for subscriptions events and subscriptions events have more limitations in how they can be filtered.
```
Comment (lidiazuin) [inline modules/ROOT/pages/security/subscriptions-authorization.adoc]:
```suggestion
Rules for subscriptions authorization are different to normal authorization rules.
Only filtering rules are available for subscriptions events and there are more limitations in how they can be filtered.
```

**Review by lidiazuin — COMMENTED** [review]
Looking good, just a few suggestions and questions.

File: modules/ROOT/pages/directives/database-mapping.adoc [hunk 9]
```diff
@@ -10,10 +10,16 @@ Each type in your GraphQL type definitions can be mapped to an entity in your Ne
 
 == `@relationship`
 
+=== Definition
+
+// tba
```
Comment (mjfwebb) [inline modules/ROOT/pages/directives/database-mapping.adoc]:
```suggestion
[source, graphql, indent=0]
----
"""
Instructs @neo4j/graphql to treat this field as a relationship. Opens up the ability to create and connect on this field.
"""
directive @relationship(
  type: String!
  """Valid and default directions for this relationship."""
  queryDirection: RelationshipQueryDirection = DEFAULT_DIRECTED
  direction: RelationshipDirection!
  """
  The name of the interface containing the properties for this relationship.
  """
  properties: String
  """
  Prevent all but these operations from being generated for this relationship
  """
  nestedOperations: [RelationshipNestedOperations!]! = [CREATE, UPDATE, DELETE, CONNECT, DISCONNECT, CONNECT_OR_CREATE]
  """Prevent aggregation for this relationship"""
  aggregate: Boolean = true
) on FIELD_DEFINITION
----
```

File: modules/ROOT/pages/directives/database-mapping.adoc [hunk 10]
```diff
@@ -70,9 +82,14 @@ Note that in addition to this type, there is an added key `properties` in the ex
 For more information, see xref::/types/relationships.adoc[Type definitions -> Relationships].
 
 
-[[type-definitions-node]]
 == `@node`
 
+=== Definition
+
+// tba
```
Comment (mjfwebb) [inline modules/ROOT/pages/directives/database-mapping.adoc]:
```suggestion
[source, graphql, indent=0]
----
"""Informs @neo4j/graphql of node metadata"""
directive @node(
  """The labels to map this GraphQL type to in the Neo4j database"""
  labels: [String!]
) on OBJECT
----
```

File: modules/ROOT/pages/directives/database-mapping.adoc [hunk 11]
```diff
@@ -217,9 +250,14 @@ await startStandaloneServer(server, {
 ----
 
 
-[[type-definitions-alias]]
 == `@alias`
 
+=== Definition
+
+// tba
```
Comment (mjfwebb) [inline modules/ROOT/pages/directives/database-mapping.adoc]:
```suggestion
[source, graphql, indent=0]
----
"""
Instructs @neo4j/graphql to map a GraphQL field to a Neo4j node or relationship property.
"""
directive @alias(
  """The name of the Neo4j property"""
  property: String!
) on FIELD_DEFINITION
----
```

File: modules/ROOT/pages/security/authentication.adoc [hunk 12]
```diff
@@ -1,14 +1,24 @@
...
 [IMPORTANT]
 ====
-Explicit authentication, configured with the `@authentication` directive, is only ever evaluated during Cypher translation time.
+Explicit authentication, configured with the `@authentication` directive, is only evaluated during Cypher translation time.
 Unauthenticated requests with queries requiring authentication never reach the database.
 ====
 
+=== Definition
+
+// tba
```
Comment (mjfwebb) [inline modules/ROOT/pages/security/authentication.adoc]:
```suggestion
```

It's too complicated to put as a definition.

Comment (mjfwebb) [reply modules/ROOT/pages/security/authentication.adoc]: There's a specific version for `validate`:

```
[source, graphql, indent=0]
----
"""
This is a simpler version of the authentication directive to be used in the validate-document step.
"""
directive @authentication(
  """operations"""
  operations: [String]
  jwt: String
) on OBJECT | FIELD_DEFINITION | SCHEMA
----
```

File: modules/ROOT/pages/security/authorization.adoc [hunk 13]
```diff
@@ -7,9 +7,19 @@ They use predicates to evaluate the data accessed by the Cypher generated from a
...
 Unauthenticated requests with queries requiring authentication never reach the database.
 
+== `@authorization`
+
+=== Definition
+
+// tba
```
Comment (mjfwebb) [inline modules/ROOT/pages/security/authorization.adoc]:
```suggestion
```

As with `authorization` this is too complicated to write a simple definition for, I think.


There exists a "simple" case for `validate`:

```
[source, graphql, indent=0]
----
"""
This is a simpler version of the authorization directive to be used in the validate-document step.
"""
directive @authorization(
  """filter"""
  filter: [String]
  """validate"""
  validate: [String]
) on OBJECT | FIELD_DEFINITION
----
```

File: modules/ROOT/pages/security/configuration.adoc [hunk 14]
```diff
@@ -146,7 +146,13 @@ By default, filtering is available on https://www.rfc-editor.org/rfc/rfc7519#sec
 
 Filtering can be configured for additional JWT claims using the `@jwt` directive and, in some circumstances, the `@jwtClaim` directive.
 
-=== The `@jwt` directive
+=== `@jwt`
+
+==== Definition
+
+// tba
```
Comment (mjfwebb) [inline modules/ROOT/pages/security/configuration.adoc]:
```suggestion
[source, graphql, indent=0]
----
"""
Instructs @neo4j/graphql that the flagged object represents the relevant JWT payload
"""
directive @jwt on OBJECT
----
```

File: modules/ROOT/pages/security/configuration.adoc [hunk 15]
```diff
@@ -163,7 +169,13 @@ The type name `JWT` is not mandatory.
 You can use any name as long as it is decorated with the `@jwt` directive.
 ====
 
-=== The `@jwtClaim` directive
+=== `@jwtClaim`
+
+==== Definition
+
+// tba
```
Comment (mjfwebb) [inline modules/ROOT/pages/security/configuration.adoc]:
```suggestion
[source, graphql, indent=0]
----
"""
Instructs @neo4j/graphql that the flagged field has a mapped path within the JWT Payload.
"""
directive @jwtClaim(
  """
  The path of the field in the real JWT as mapped within the JWT Payload.
  """
  path: String!
) on FIELD_DEFINITION
----
```

File: modules/ROOT/pages/security/subscriptions-authorization.adoc [hunk 16]
```diff
@@ -3,13 +3,22 @@
...
 
 All subscriptions authorization rules have an implied requirement for authentication, given that the rules are normally evaluated against values in the JWT payload.
 
-== Rules
+=== Definition
+
+// tba
```
Comment (mjfwebb) [inline modules/ROOT/pages/security/subscriptions-authorization.adoc]:
```suggestion
```

Same as others.

```
[source, graphql, indent=0]
----
"""
This is a simpler version of the subscriptionsAuthorization directive to be used in the validate-document step.
"""
directive @subscriptionsAuthorization(
  """filter"""
  filter: [String]!
) on OBJECT | FIELD_DEFINITION
----
```

File: modules/ROOT/pages/directives/schema-configuration/type-configuration.adoc [hunk 17]
```diff
@@ -184,9 +184,14 @@ type SomeType {
 ----
 
 
-[[type-definitions-plural]]
 == `@plural`
 
+=== Definition
+
+// tba
```
Comment (mjfwebb) [inline modules/ROOT/pages/directives/schema-configuration/type-configuration.adoc]:
```suggestion
[source, graphql, indent=0]
----
"""
Instructs @neo4j/graphql to use the given value as the plural of the type name
"""
directive @plural(
  """The value to use as the plural of the type name."""
  value: String!
) on OBJECT | INTERFACE | UNION
----
```

File: modules/ROOT/pages/directives/database-mapping.adoc [hunk 18]
```diff
@@ -28,8 +34,14 @@ type Actor {
 }
 ----
 
-Note that, in this case, there is a directive on each "end" of the relationship, but it is not essential.
+[NOTE]
+====
+The `@relationship` directive is used twice, once on each end of the relationship.
+This is the standard way of modeling a relationship with the GraphQL library.
```
Comment (rsill-neo4j) [inline modules/ROOT/pages/directives/database-mapping.adoc]:
```suggestion
This is the standard way of modeling a relationship with the GraphQL Library.
```

File: modules/ROOT/pages/directives/indexes-and-constraints.adoc (hunk 4 above)
Comment (rsill-neo4j) [reply modules/ROOT/pages/directives/indexes-and-constraints.adoc]: it was in reverse, basically - 
i changed the heading text, and added the custom TOC headers to preserve the original URLs which are used throughout the docs.

i could instead update the links throughout the docs, if that's what we usually do

File: modules/ROOT/pages/directives/indexes-and-constraints.adoc [hunk 19]
```diff
@@ -221,6 +165,66 @@ query {
...
+----
+"""Informs @neo4j/graphql that there should be a uniqueness constraint in the database for the decorated field."""
+directive @unique(
+    """The name which should be used for this constraint. By default; type name, followed by an underscore, followed by the field name."""
+    constraintName: String
+) on FIELD_DEFINITION
+----
+
+Using this directive does not automatically ensure the existence of these constraints, and you will need to run a function on server startup. 
+See the section xref::/directives/indexes-and-constraints.adoc#_asserting_constraints[Asserting constraints] for details.
```
Comment (rsill-neo4j) [inline modules/ROOT/pages/directives/indexes-and-constraints.adoc]:
```suggestion
See section xref::/directives/indexes-and-constraints.adoc#_asserting_constraints[Asserting constraints] for details.
```

**Comment by neo-technology-commit-status-publisher** [comment]
Thanks for the documentation updates.

The preview documentation has now been torn down - reopening this PR will republish it.

**Comment by neo-technology-commit-status-publisher** [comment]
## 💔 All backports failed

| Status | Branch | Result |
|:------:|:------:|:------|
|❌|6.x|**Backport failed because of merge conflicts**<br><br>You might need to backport the following PRs to 6.x:<br> - [wording changes, added a link](https://github.com/neo4j/docs-graphql/pull/203)|
|❌|7.x|Backport failed because of merge conflicts|

### Manual backport
To create the backport manually run:
```
backport --pr 211
```

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport) and see the [Github Action logs](https://github.com/neo4j/docs-graphql/actions/runs/11739155562) for details

**Pull request merged**